subsystem/kubeapi_test.go

```diff
@@ -0,0 +1,305 @@
+package subsystem
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/go-openapi/swag"
+	"github.com/jinzhu/gorm"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	"github.com/openshift/assisted-service/models"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+func deployPullSecretResource(ctx context.Context, client k8sclient.Client, name, secret string) {
+	data := map[string][]byte{"pullSecret": []byte(secret)}
+	s := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: Options.Namespace,
+			Name:      name,
+		},
+		Data: data,
+	}
+	Expect(client.Create(ctx, s)).ShouldNot(HaveOccurred())
+}
+
+func deployClusterCRD(ctx context.Context, client k8sclient.Client, spec *v1alpha1.ClusterSpec) {
+	err := client.Create(ctx, &v1alpha1.Cluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Cluster",
+			APIVersion: getAPIVersion(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: Options.Namespace,
+			Name:      spec.Name,
+		},
+		Spec: *spec,
+	})
+	Expect(err).ShouldNot(HaveOccurred())
+}
+
+func getAPIVersion() string {
+	return fmt.Sprintf("%s/%s", v1alpha1.GroupVersion.Group, v1alpha1.GroupVersion.Version)
+}
+
+func getClusterFromDB(
+	ctx context.Context, client k8sclient.Client, db *gorm.DB, key types.NamespacedName, timeout int) *common.Cluster {
+
+	var err error
+	cluster := &common.Cluster{}
+	start := time.Now()
+	for time.Duration(timeout)*time.Second > time.Since(start) {
+		err = db.Take(&cluster, "kube_key_name = ? and kube_key_namespace = ?", key.Name, key.Namespace).Error
+		if err == nil {
+			return cluster
+		}
+		if !gorm.IsRecordNotFoundError(err) {
+			Expect(err).ShouldNot(HaveOccurred())
+		}
+		clusterCRD := getClusterCRD(ctx, client, key)
+		Expect(clusterCRD.Status.Error).Should(Equal(""))
+		time.Sleep(time.Second)
+	}
+	Expect(err).ShouldNot(HaveOccurred())
+	return cluster
+}
+
+func getClusterCRD(ctx context.Context, client k8sclient.Client, key types.NamespacedName) *v1alpha1.Cluster {
+	cluster := &v1alpha1.Cluster{}
+	err := client.Get(ctx, key, cluster)
+	Expect(err).ShouldNot(HaveOccurred())
+	return cluster
+}
+
+func updateClusterCRD(
```

Comment:
I think that it's not used?

Reply:
it was used by the tests I removed, will remove it as well